Add reminder about docstrings, objects, and intervening lines

ubertakter opened the pull request:
Added a specific call out (as a reminder note) that no lines are allowed between docstrings and the object being documented. This is strictly an addition, with none of the current text being changed.

ubertakter committed:
Update documentation.md

Update documentation to include a note in point 8 on making sure there are no lines between the end of a docstring and the start of the function declaration.

ubertakter committed:
Update documentation.md

Added a reminder to emphasize no lines allowed between docstring and documented object.

ubertakter committed:
Merge branch 'JuliaLang:master' into master

ubertakter committed:
Update documentation.md

Updated to remove some some stubborn whitespace.

nsajko commented on doc/src/manual/documentation.md:
````diff
@@ -37,6 +37,8 @@ the documented object. Here is a basic example:
 foo(xs::Array) = ...
 ```
+!!! note "Reminder"
+    Having any lines between the docstring and the object being documented (function, macro, type or instance), while human readable, are not allowed and may break automated documentation tools.
````
Disliking the current wording:

* `(function, macro, type or instance)`: seems redundant

* `while human readable`: subjective and unnecessary

* `may break automated documentation tools`: seems to weak/imprecise

I'm too tired to make a completely thoughtful suggestion, however maybe something like this would be an improvement?

```suggestion
    Any empty lines between the docstring and the object being documented detach the former from the latter, making the docstring useless.
```

ubertakter commented:
I made it redundant on purpose, though I don't have a problem moving the
"above" sentence into an admonition.

I like your changes though I don't like calling documentation useless 😁.
How about instead of "making the docstring useless" something like "likely
breaking documentation tools".

On Tue, May 27, 2025, 18:55 Neven Sajko ***@***.***> wrote:

> *nsajko* left a comment (JuliaLang/julia#58541)
> <https://github.com/JuliaLang/julia/pull/58541#issuecomment-2914462909>
>
> A little bit above your change is this sentence that makes your change
> redundant:
>
> Note that no blank lines or comments may intervene between a docstring and
> the documented object.
>
> Maybe it'd be better to just move that sentence into an admonition,
> without the "note that" prefix?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/JuliaLang/julia/pull/58541#issuecomment-2914462909>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AEBNIEYW26DHQPVXENXRBF33AT3O3AVCNFSM6AAAAAB6AY3S4GVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDSMJUGQ3DEOJQHE>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

LilithHafner reviewed (commented):
Thanks for working on these docs!

ubertakter committed:
Fixed bullet numbering in "Writing Documentation"

ubertakter commented on doc/src/manual/documentation.md:
```diff
@@ -242,16 +244,17 @@ As in the example above, we recommend following some simple conventions when wri
    This makes it clearer where docstrings start and end.
-9. Respect the line length limit used in the surrounding code.
+   
+10. Respect the line length limit used in the surrounding code.
```
Definitely unintentional.

ubertakter committed:
Updated to use suggested text

Updated to use text suggested by @nsajko because apparently I can't work GitHub correctly or something.

ubertakter committed:
Merge branch 'master' into master

nsajko commented on doc/src/manual/documentation.md:
````diff
@@ -242,6 +244,7 @@ As in the example above, we recommend following some simple conventions when wri
    ```
 
    This makes it clearer where docstrings start and end.
+
````
Unintentionally added empty line here:

```suggestion
```

ubertakter committed:
Update doc/src/manual/documentation.md

Sorry about that. I'm traveling now, hopefully I don't mess something up by doing this on my phone.

Co-authored-by: Neven Sajko <4944410+nsajko@users.noreply.github.com>

LilithHafner commented:
Thanks you @ubertakter and @nsajko!
